[Automation] native-image run fails for org.jboss:jandex:2.4.5.Final

## native-image run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.4.5.Final" ./gradlew clean test -Pcoordinates="org.jboss:jandex:2.4.2.Final"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/24753645127/job/72422175831
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
5.0.3+9.1/lib/svm/builder/svm-enterprise.jar:/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/builder/svm-configure.jar \
-Dsvm.modulesupport.addedModules=java.rmi,java.sql,org.graalvm.collections,jdk.jdi,java.logging,java.xml.crypto,java.transaction.xa,java.xml,jdk.jfr,jdk.internal.vm.ci,java.datatransfer,org.graalvm.truffle.compiler,jdk.internal.md,java.desktop,java.naming,jdk.internal.ed,java.prefs,java.net.http,jdk.compiler,java.security.sasl,jdk.internal.opt,org.graalvm.nativeimage,jdk.zipfs,jdk.attach,java.base,jdk.management.agent,org.graalvm.word,jdk.internal.le,java.management,java.sql.rowset,jdk.unsupported,jdk.jlink,jdk.jdwp.agent,jdk.internal.jvmstat,java.instrument,java.security.jgss,jdk.management,java.compiler,java.scripting,jdk.graal.compiler,jdk.jdeps,java.management.rmi \
--add-modules=java.rmi,java.sql,org.graalvm.collections,jdk.jdi,java.logging,java.xml.crypto,java.transaction.xa,java.xml,jdk.jfr,jdk.internal.vm.ci,java.datatransfer,org.graalvm.truffle.compiler,jdk.internal.md,java.desktop,java.naming,jdk.internal.ed,java.prefs,java.net.http,jdk.compiler,java.security.sasl,jdk.internal.opt,org.graalvm.nativeimage,jdk.zipfs,jdk.attach,java.base,jdk.management.agent,org.graalvm.word,jdk.internal.le,java.management,java.sql.rowset,jdk.unsupported,jdk.jlink,jdk.jdwp.agent,jdk.internal.jvmstat,java.instrument,java.security.jgss,jdk.management,java.compiler,java.scripting,jdk.graal.compiler,jdk.jdeps,java.management.rmi \
--module \
org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner \
-keepalive \
/proc/7428/comm \
-imagecp \
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/libs/org.jboss.jandex_tests.jar:/home/runner/.m2/repository/org/graalvm/buildtools/junit-platform-native/1.0.1-SNAPSHOT/junit-platform-native-1.0.1-SNAPSHOT.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter/5.9.2/26c586fbe0ebd81b48c9f11f0d998124248697ae/junit-jupiter-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-reporting/1.9.2/8bea292e2a546e4c21071f6bee60917ffd9f0101/junit-platform-reporting-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-launcher/1.9.2/38e0bfad8c57d4cd1a8f27926c25ffe9543068d6/junit-platform-launcher-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.vintage/junit-vintage-engine/5.9.2/53421816bde124a564a64ba005dcc0c8e66a9722/junit-vintage-engine-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-params/5.9.2/bc2765afb7b85b583c710dd259a11c6b8c39e912/junit-jupiter-params-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-engine/5.9.2/572f7a553b53f83ee59cc045ce1c3772864ab76c/junit-jupiter-engine-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-api/5.9.2/fed843581520eac594bc36bb4b0f55e7b947dda9/junit-jupiter-api-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-engine/1.9.2/40aeef2be7b04f96bb91e8b054affc28b7c7c935/junit-platform-engine-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-commons/1.9.2/6f9f8621d8230cd38aa42e58ccbc0c00569131ce/junit-platform-commons-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-console/1.9.2/5548148dd55e820c42bd817ab93c1eaa05ce37f6/junit-platform-console-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.jboss/jandex/2.4.5.Final/9b4634d1fa28628549eb986b7c0c73f387090fba/jandex-2.4.5.Final.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.assertj/assertj-core/3.22.0/c300c0c6a24559f35fa0bd3a5472dc1edcd0111e/assertj-core-3.22.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/junit/junit/4.13.2/8ac9e16d933b6fb43bc7f576336b8f4d7eb5ba12/junit-4.13.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.opentest4j/opentest4j/1.2.0/28c11eb91f9b6d8e200631d46e20a7f407f2a046/opentest4j-1.2.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.hamcrest/hamcrest-core/1.3/42a25dc3219429f0e5d060061f71acb49bf010a0/hamcrest-core-1.3.jar:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/classes/java/test:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/resources/test:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/test-results/test/testlist \
-imagemp \
/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar \
-H:CLibraryPath=/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/clibraries/linux-amd64/glibc,/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/clibraries/linux-amd64,/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/clibraries \
-H:Path@driver=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/native/nativeTestCompile \
-H:BuildOutputJSONFile@user=/tmp/native-image-build-output.json \
-H:FallbackThreshold@user+api=0 \
-H:Name@user+api=org.jboss.jandex_tests-tests \
-H:ConfigurationFileDirectories@user=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/native/generated/generateTestResourcesConfigFile,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.jboss/jandex/2.4.2.Final \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=5d5b4e8c-6f16-71f5-8fbe-a89a48a4c1a8 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature,com.oracle.svm.polyglot.scala.ScalaFeature'
]
========================================================================================================================
GraalVM Native Image: Generating 'org.jboss.jandex_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                    (5.8s @ 0.15GB)
 Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.0.3+9.1
 Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 C compiler: gcc (linux, x86_64, 13.3.0)
 Garbage collector: Serial GC (max heap size: 80% of RAM)
 2 user-specific feature(s):
 - com.oracle.svm.thirdparty.gson.GsonFeature
 - org.graalvm.junit.platform.JUnitPlatformFeature
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 12.67GB of memory (75.6% of system memory, $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                   (32.6s @ 0.82GB)
    7,936 types,   9,071 fields, and  35,325 methods found reachable
    2,547 types,     106 fields, and   1,229 methods registered for reflection
       59 types,      60 fields, and      52 methods registered for JNI access
        0 downcalls and 0 upcalls registered for foreign access
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                               (4.0s @ 0.94GB)
[4/8] Parsing methods...      [**]                                                                       (4.2s @ 0.64GB)
[5/8] Inlining methods...     [***]                                                                      (2.1s @ 0.70GB)
[6/8] Compiling methods...    [****]                                                                    (17.7s @ 0.84GB)
[7/8] Laying out methods...   [**]                                                                       (3.4s @ 0.82GB)
[8/8] Creating image...       [**]                                                                       (3.1s @ 0.94GB)
   9.23MB (35.99%) for code area:    21,944 compilation units
  12.78MB (49.84%) for image heap:  204,740 objects and 151 resources
   3.63MB (14.17%) for other data
  25.64MB in total image size, 23.66MB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   4.54MB java.base                                            2.68MB byte[] for code metadata
   2.07MB svm.jar (Native Image)                               2.46MB byte[] for java.lang.String
 556.57kB assertj-core-3.22.0.jar                              1.30MB java.lang.String
 481.39kB java.xml                                           991.27kB java.lang.Class
 197.65kB junit-jupiter-engine-5.9.2.jar                     825.34kB com.oracle.svm.core.hub.DynamicHubCompanion
 157.32kB jandex-2.4.5.Final.jar                             698.66kB byte[] for general heap data
 121.98kB junit-platform-launcher-1.9.2.jar                  449.98kB java.util.concurrent.ConcurrentHashMap$Node
 103.85kB java.logging                                       363.91kB heap alignment
  99.33kB junit-platform-engine-1.9.2.jar                    299.58kB java.util.HashMap$Node
  82.59kB junit-4.13.2.jar                                   271.02kB java.lang.Object[]
 474.97kB for 21 more packages                                 2.44MB for 1995 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 17 component(s) is embedded in binary (911B). 4 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                        5.2s (6.8% of total time) in 689 GCs | Peak RSS: 1.73GB | CPU load: 3.63
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/native/nativeTestCompile/org.jboss.jandex_tests-tests (executable)
 /tmp/native-image-build-output.json (build_info)
========================================================================================================================
Finished generating 'org.jboss.jandex_tests-tests' in 1m 14s.
[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

org_jboss.jandex.IndexerTest > indexesLoadedClassFromItsClassResource() FAILED

org_jboss.jandex.StrongInternPoolTest > serializesAndDeserializesInternedEntries() SUCCESSFUL


Failures (1):
  JUnit Jupiter:IndexerTest:indexesLoadedClassFromItsClassResource()
    MethodSource [className = 'org_jboss.jandex.IndexerTest', methodName = 'indexesLoadedClassFromItsClassResource', methodParameterTypes = '']
    => org.graalvm.nativeimage.MissingReflectionRegistrationError: Cannot reflectively instantiate the array class 'org.jboss.jandex.ClassInfo[]'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

  {
    "type": "org.jboss.jandex.ClassInfo[]"
  }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.reportArrayInstantiation(MissingReflectionRegistrationUtils.java:145)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.graal.snippets.SubstrateAllocationSnippets.arrayHubErrorStub(SubstrateAllocationSnippets.java:422)
       org.jboss.jandex.Utils.unfold(Utils.java:56)
       org.jboss.jandex.Index.create(Index.java:142)
       org.jboss.jandex.Indexer.complete(Indexer.java:2032)
       org_jboss.jandex.IndexerTest.indexesLoadedClassFromItsClassResource(IndexerTest.java:29)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       [...]

Test run finished after 4 ms
[         4 containers found      ]
[         0 containers skipped    ]
[         4 containers started    ]
[         0 containers aborted    ]
[         4 containers successful ]
[         0 containers failed     ]
[         2 tests found           ]

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/native/nativeTestCompile/org.jboss.jandex_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 18s
[         0 tests skipped         ]
[         2 tests started         ]
[         0 tests aborted         ]
[         1 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 6 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.jboss:jandex:2.4.2.Final failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 19s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 2.4.5.Final failed with exit code 1.
FAILED[native-image run][2.4.5.Final][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.4.5.Final" ./gradlew clean test -Pcoordinates="org.jboss:jandex:2.4.2.Final"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.4.5.Final" ./gradlew clean test -Pcoordinates="org.jboss:jandex:2.4.2.Final"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/24753645127/job/72422175809
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
ache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/vmaccess.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/pointsto.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/native-image-base.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/espresso-svm.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/svm.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/svm-capnproto-runtime.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/svm-enterprise.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/svm-configure.jar \
--module \
org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner \
-keepalive \
/proc/7383/comm \
-imagecp \
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/libs/org.jboss.jandex_tests.jar:/home/runner/.m2/repository/org/graalvm/buildtools/junit-platform-native/1.0.1-SNAPSHOT/junit-platform-native-1.0.1-SNAPSHOT.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter/5.9.2/26c586fbe0ebd81b48c9f11f0d998124248697ae/junit-jupiter-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-reporting/1.9.2/8bea292e2a546e4c21071f6bee60917ffd9f0101/junit-platform-reporting-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-launcher/1.9.2/38e0bfad8c57d4cd1a8f27926c25ffe9543068d6/junit-platform-launcher-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.vintage/junit-vintage-engine/5.9.2/53421816bde124a564a64ba005dcc0c8e66a9722/junit-vintage-engine-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-params/5.9.2/bc2765afb7b85b583c710dd259a11c6b8c39e912/junit-jupiter-params-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-engine/5.9.2/572f7a553b53f83ee59cc045ce1c3772864ab76c/junit-jupiter-engine-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-api/5.9.2/fed843581520eac594bc36bb4b0f55e7b947dda9/junit-jupiter-api-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-engine/1.9.2/40aeef2be7b04f96bb91e8b054affc28b7c7c935/junit-platform-engine-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-commons/1.9.2/6f9f8621d8230cd38aa42e58ccbc0c00569131ce/junit-platform-commons-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-console/1.9.2/5548148dd55e820c42bd817ab93c1eaa05ce37f6/junit-platform-console-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.jboss/jandex/2.4.5.Final/9b4634d1fa28628549eb986b7c0c73f387090fba/jandex-2.4.5.Final.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.assertj/assertj-core/3.22.0/c300c0c6a24559f35fa0bd3a5472dc1edcd0111e/assertj-core-3.22.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/junit/junit/4.13.2/8ac9e16d933b6fb43bc7f576336b8f4d7eb5ba12/junit-4.13.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.opentest4j/opentest4j/1.2.0/28c11eb91f9b6d8e200631d46e20a7f407f2a046/opentest4j-1.2.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.hamcrest/hamcrest-core/1.3/42a25dc3219429f0e5d060061f71acb49bf010a0/hamcrest-core-1.3.jar:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/classes/java/test:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/resources/test:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/test-results/test/testlist \
-imagemp \
/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar \
-H:CLibraryPath=/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/clibraries/linux-amd64/glibc,/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/clibraries/linux-amd64,/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/clibraries \
-H:Path@driver=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/native/nativeTestCompile \
-H:BuildOutputJSONFile@user=/tmp/native-image-build-output.json \
-H:FallbackThreshold@user+api=0 \
-H:Name@user+api=org.jboss.jandex_tests-tests \
-H:ConfigurationFileDirectories@user=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/native/generated/generateTestResourcesConfigFile,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.jboss/jandex/2.4.2.Final \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=5d5b4e8c-6f16-71f5-8fbe-a89a48a4c1a8 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=4
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'org.jboss.jandex_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (6.7s @ 0.21GiB)
 Builder configuration:
 - Java version: 25.0.2+10-LTS, vendor version: Oracle GraalVM 25.1.0-dev+10.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [******]                                                                 (33.1s @ 0.83GiB)
    8,063 types,   9,132 fields, and  35,723 methods found reachable
      283 types,      95 fields, and     913 methods registered for reflection
       59 types,      60 fields, and      52 methods registered for JNI access
      151 resource accesses registered with 262.75KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (4.1s @ 0.93GiB)
[4/8] Parsing methods...      [**]                                                                      (3.4s @ 0.97GiB)
[5/8] Inlining methods...     [***]                                                                     (2.6s @ 0.69GiB)
[6/8] Compiling methods...    [****]                                                                   (18.6s @ 0.79GiB)
[7/8] Laying out methods...   [**]                                                                      (3.6s @ 0.81GiB)
[8/8] Creating image...       [**]                                                                      (3.3s @ 0.91GiB)
   8.17MiB (33.56%) for code area:    22,187 compilation units
  12.69MiB (52.10%) for image heap:  213,900 objects and 21 resources
   3.49MiB (14.34%) for other data
  24.35MiB in total image size, 22.44MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
 916.82KiB o.g.n.runtime/com.oracle.svm.core.code              2.52MiB byte[] for code metadata
 696.59KiB java.base/java.util                                 2.43MiB byte[] for string data
 407.38KiB java.base/jdk.internal.classfile.impl               1.31MiB java.base/java.lang.String
 317.88KiB java.base/java.lang                               978.63KiB java.base/java.lang.Class
 228.80KiB java.base/java.text                               881.89KiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 227.84KiB java.base/java.io                                 513.30KiB byte[] for general heap data
 224.55KiB java.base/com.sun.crypto.provider                 452.03KiB java.base/java.util.HashMap$Node
 223.93KiB org.assertj.core.internal.bytebuddy.jar.asm       394.68KiB heap alignment
 221.73KiB java.base/java.util.concurrent                    312.58KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 168.96KiB java.base/java.lang.invoke                        269.36KiB byte[] for reflection metadata
   4.62MiB for 303 more packages                               2.71MiB for 2008 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 17 component(s) is embedded in binary (991B). 4 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       5.2s (6.6% of total time) in 693 GCs | Peak RSS: 1.57GiB | CPU load: 3.59
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/native/nativeTestCompile/org.jboss.jandex_tests-tests (executable, 22.44MiB)
 /tmp/native-image-build-output.json (build_info, 1.02KiB)
========================================================================================================================
Finished generating 'org.jboss.jandex_tests-tests' in 1m 17s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

org_jboss.jandex.IndexerTest > indexesLoadedClassFromItsClassResource() FAILED

org_jboss.jandex.StrongInternPoolTest > serializesAndDeserializesInternedEntries() SUCCESSFUL


Failures (1):
  JUnit Jupiter:IndexerTest:indexesLoadedClassFromItsClassResource()
    MethodSource [className = 'org_jboss.jandex.IndexerTest', methodName = 'indexesLoadedClassFromItsClassResource', methodParameterTypes = '']
    => org.graalvm.nativeimage.MissingReflectionRegistrationError: Cannot reflectively instantiate the array class 'org.jboss.jandex.ClassInfo[]'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

    {
      "type": "org.jboss.jandex.ClassInfo[]"
    }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.reportArrayInstantiation(MissingReflectionRegistrationUtils.java:145)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.graal.snippets.SubstrateAllocationSnippets.arrayHubErrorStub(SubstrateAllocationSnippets.java:437)
       org.jboss.jandex.Utils.unfold(Utils.java:56)
       org.jboss.jandex.Index.create(Index.java:142)
       org.jboss.jandex.Indexer.complete(Indexer.java:2032)
       org_jboss.jandex.IndexerTest.indexesLoadedClassFromItsClassResource(IndexerTest.java:29)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       [...]

Test run finished after 2 ms
[         4 containers found      ]
[         0 containers skipped    ]
[         4 containers started    ]
[         0 containers aborted    ]
[         4 containers successful ]
[         0 containers failed     ]
[         2 tests found           ]
[         0 tests skipped         ]
[         2 tests started         ]
[         0 tests aborted         ]
[         1 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/native/nativeTestCompile/org.jboss.jandex_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan
 to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 21s
14 actionable tasks: 6 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.jboss:jandex:2.4.2.Final failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 23s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 2.4.5.Final failed with exit code 1.
FAILED[native-image run][2.4.5.Final][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.4.5.Final" ./gradlew clean test -Pcoordinates="org.jboss:jandex:2.4.2.Final"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.4.5.Final" ./gradlew clean test -Pcoordinates="org.jboss:jandex:2.4.2.Final"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/24753645127/job/72422175845
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
che/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/vmaccess.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/pointsto.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/native-image-base.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/espresso-svm.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/svm.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/svm-capnproto-runtime.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/svm-enterprise.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/builder/svm-configure.jar \
--module \
org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner \
-keepalive \
/proc/7310/comm \
-imagecp \
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/libs/org.jboss.jandex_tests.jar:/home/runner/.m2/repository/org/graalvm/buildtools/junit-platform-native/1.0.1-SNAPSHOT/junit-platform-native-1.0.1-SNAPSHOT.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter/5.9.2/26c586fbe0ebd81b48c9f11f0d998124248697ae/junit-jupiter-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-reporting/1.9.2/8bea292e2a546e4c21071f6bee60917ffd9f0101/junit-platform-reporting-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-launcher/1.9.2/38e0bfad8c57d4cd1a8f27926c25ffe9543068d6/junit-platform-launcher-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.vintage/junit-vintage-engine/5.9.2/53421816bde124a564a64ba005dcc0c8e66a9722/junit-vintage-engine-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-params/5.9.2/bc2765afb7b85b583c710dd259a11c6b8c39e912/junit-jupiter-params-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-engine/5.9.2/572f7a553b53f83ee59cc045ce1c3772864ab76c/junit-jupiter-engine-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-api/5.9.2/fed843581520eac594bc36bb4b0f55e7b947dda9/junit-jupiter-api-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-engine/1.9.2/40aeef2be7b04f96bb91e8b054affc28b7c7c935/junit-platform-engine-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-commons/1.9.2/6f9f8621d8230cd38aa42e58ccbc0c00569131ce/junit-platform-commons-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-console/1.9.2/5548148dd55e820c42bd817ab93c1eaa05ce37f6/junit-platform-console-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.jboss/jandex/2.4.5.Final/9b4634d1fa28628549eb986b7c0c73f387090fba/jandex-2.4.5.Final.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.assertj/assertj-core/3.22.0/c300c0c6a24559f35fa0bd3a5472dc1edcd0111e/assertj-core-3.22.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/junit/junit/4.13.2/8ac9e16d933b6fb43bc7f576336b8f4d7eb5ba12/junit-4.13.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.opentest4j/opentest4j/1.2.0/28c11eb91f9b6d8e200631d46e20a7f407f2a046/opentest4j-1.2.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.hamcrest/hamcrest-core/1.3/42a25dc3219429f0e5d060061f71acb49bf010a0/hamcrest-core-1.3.jar:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/classes/java/test:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/resources/test:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/test-results/test/testlist \
-imagemp \
/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar \
-H:CLibraryPath=/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/clibraries/linux-amd64/glibc,/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/clibraries/linux-amd64,/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/clibraries \
-H:Path@driver=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/native/nativeTestCompile \
-H:BuildOutputJSONFile@user=/tmp/native-image-build-output.json \
-H:FallbackThreshold@user+api=0 \
-H:Name@user+api=org.jboss.jandex_tests-tests \
-H:ConfigurationFileDirectories@user=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/native/generated/generateTestResourcesConfigFile,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.jboss/jandex/2.4.2.Final \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:FutureDefaults@user+api=all \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=3bad17d2-2ebf-4b90-6151-96826000f84c \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=4
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'org.jboss.jandex_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (6.7s @ 0.21GiB)
 Builder configuration:
 - Java version: 25.0.2+10-LTS, vendor version: Oracle GraalVM 25.1.0-dev+10.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (36.0s @ 0.89GiB)
    8,606 types,   9,934 fields, and  38,767 methods found reachable
      411 types,      95 fields, and   1,119 methods registered for reflection
       59 types,      60 fields, and      52 methods registered for JNI access
      153 resource accesses registered with 262.89KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (4.3s @ 1.03GiB)
[4/8] Parsing methods...      [**]                                                                      (4.4s @ 0.73GiB)
[5/8] Inlining methods...     [***]                                                                     (2.0s @ 0.80GiB)
[6/8] Compiling methods...    [****]                                                                   (20.7s @ 0.79GiB)
[7/8] Laying out methods...   [**]                                                                      (3.0s @ 0.97GiB)
[8/8] Creating image...       [**]                                                                      (3.7s @ 1.08GiB)
   9.30MiB (33.71%) for code area:    24,509 compilation units
  14.50MiB (52.56%) for image heap:  252,216 objects and 23 resources
   3.79MiB (13.74%) for other data
  27.59MiB in total image size, 25.50MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.02MiB o.g.n.runtime/com.oracle.svm.core.code              2.87MiB byte[] for code metadata
 703.82KiB java.base/java.util                                 2.73MiB byte[] for string data
 407.34KiB java.base/jdk.internal.classfile.impl               1.45MiB java.base/java.lang.String
 327.24KiB java.base/java.lang                                 1.03MiB java.base/java.lang.Class
 267.01KiB java.base/com.sun.crypto.provider                 941.28KiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 241.82KiB java.base/java.io                                 877.91KiB byte[] for general heap data
 231.37KiB java.base/java.text                               455.88KiB java.base/java.util.HashMap$Node
 223.93KiB org.assertj.core.internal.bytebuddy.jar.asm       337.57KiB byte[] for reflection metadata
 221.85KiB java.base/java.util.concurrent                    333.79KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 192.83KiB java.base/sun.security.provider                   308.47KiB java.base/j.u.concurrent.ConcurrentHashMap$Node
   5.52MiB for 313 more packages                               3.25MiB for 2164 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 17 component(s) is embedded in binary (994B). 4 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       5.8s (6.9% of total time) in 802 GCs | Peak RSS: 1.70GiB | CPU load: 3.62
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/native/nativeTestCompile/org.jboss.jandex_tests-tests (executable, 25.50MiB)
 /tmp/native-image-build-output.json (build_info, 1.02KiB)
========================================================================================================================
Finished generating 'org.jboss.jandex_tests-tests' in 1m 23s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/native/nativeTestCompile

> Task :nativeTest FAILED
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/native/nativeTestCompile/org.jboss.jandex_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 26s

[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

org_jboss.jandex.IndexerTest > indexesLoadedClassFromItsClassResource() FAILED

org_jboss.jandex.StrongInternPoolTest > serializesAndDeserializesInternedEntries() SUCCESSFUL


Failures (1):
  JUnit Jupiter:IndexerTest:indexesLoadedClassFromItsClassResource()
    MethodSource [className = 'org_jboss.jandex.IndexerTest', methodName = 'indexesLoadedClassFromItsClassResource', methodParameterTypes = '']
    => org.graalvm.nativeimage.MissingReflectionRegistrationError: Cannot reflectively instantiate the array class 'org.jboss.jandex.ClassInfo[]'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

    {
      "type": "org.jboss.jandex.ClassInfo[]"
    }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.reportArrayInstantiation(MissingReflectionRegistrationUtils.java:145)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.graal.snippets.SubstrateAllocationSnippets.arrayHubErrorStub(SubstrateAllocationSnippets.java:437)
       org.jboss.jandex.Utils.unfold(Utils.java:56)
       org.jboss.jandex.Index.create(Index.java:142)
       org.jboss.jandex.Indexer.complete(Indexer.java:2032)
       org_jboss.jandex.IndexerTest.indexesLoadedClassFromItsClassResource(IndexerTest.java:29)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       [...]

Test run finished after 2 ms
[         4 containers found      ]
[         0 containers skipped    ]
[         4 containers started    ]
[         0 containers aborted    ]
[         4 containers successful ]
[         0 containers failed     ]
[         2 tests found           ]
[         0 tests skipped         ]
[         2 tests started         ]
[         0 tests aborted         ]
[         1 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.jboss/jandex/2.4.2.Final/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 6 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.jboss:jandex:2.4.2.Final failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 28s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 2.4.5.Final failed with exit code 1.
FAILED[native-image run][2.4.5.Final][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.4.5.Final" ./gradlew clean test -Pcoordinates="org.jboss:jandex:2.4.2.Final"]
```

--- a/metadata/org.jboss/jandex/2.4.5.Final/reachability-metadata.json
+++ b/metadata/org.jboss/jandex/2.4.5.Final/reachability-metadata.json
@@ -1,0 +1,38 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "org.jboss.jandex.StrongInternPool"
+      },
+      "type": "java.lang.String",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "org.jboss.jandex.StrongInternPool"
+      },
+      "type": "org.jboss.jandex.StrongInternPool",
+      "serializable": true
+    },
+    {
+      "condition": {
+        "typeReached": "org.jboss.jandex.Utils"
+      },
+      "type": "org.jboss.jandex.AnnotationInstance[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.jboss.jandex.Utils"
+      },
+      "type": "org.jboss.jandex.ClassInfo[]"
+    }
+  ],
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "org.jboss.jandex.Indexer"
+      },
+      "glob": "org_jboss/jandex/IndexerTest$IndexedSample.class"
+    }
+  ]
+}

--- a/metadata/org.jboss/jandex/index.json
+++ b/metadata/org.jboss/jandex/index.json
@@ -1,6 +1,21 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "2.4.5.Final",
+    "test-version" : "2.4.2.Final",
+    "tested-versions" : [
+      "2.4.5.Final"
+    ],
+    "source-code-url" : "https://repo.maven.apache.org/maven2/org/jboss/jandex/$version$/jandex-$version$-sources.jar",
+    "repository-url" : "https://github.com/smallrye/jandex",
+    "test-code-url" : "https://github.com/smallrye/jandex/tree/$version$/src/test/java",
+    "documentation-url" : "https://github.com/smallrye/jandex/blob/$version$/README.md",
+    "description" : "Jandex indexes Java class metadata and annotations into a compact structure for fast offline lookup. It also provides APIs and tools to query, persist, and reload that index without relying on runtime reflection.",
+    "allowed-packages" : [
+      "org.jboss.jandex"
+    ]
+  },
+  {
     "metadata-version" : "2.4.2.Final",
     "source-code-url" : "https://repo1.maven.org/maven2/org/jboss/jandex/$version$/jandex-$version$-sources.jar",
     "repository-url" : "https://github.com/smallrye/jandex",

--- a/stats/org.jboss/jandex/2.4.5.Final/execution-metrics.json
+++ b/stats/org.jboss/jandex/2.4.5.Final/execution-metrics.json
@@ -1,0 +1,137 @@
+{
+  "fix_ni_run:2026-06-10": {
+    "timestamp": "2026-06-10T02:23:20.215377Z",
+    "library": "org.jboss:jandex:2.4.5.Final",
+    "starting_commit": "cf46581ffcef7b574d89b113c913b906b202f63b",
+    "ending_commit": "5ada4d9473928567a3cdfad1db70e086cdc3cc26",
+    "previous_library": "org.jboss:jandex:2.4.2.Final",
+    "strategy_name": "library_update_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "2.4.5.Final",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 3,
+            "totalCalls": 3
+          },
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 4,
+        "totalCalls": 4
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 4696,
+          "missed": 23324,
+          "ratio": 0.167595,
+          "total": 28020
+        },
+        "line": {
+          "covered": 836,
+          "missed": 4950,
+          "ratio": 0.144487,
+          "total": 5786
+        },
+        "method": {
+          "covered": 146,
+          "missed": 1059,
+          "ratio": 0.121162,
+          "total": 1205
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "2.4.2.Final",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 2,
+            "totalCalls": 2
+          },
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 3,
+        "totalCalls": 3
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 4650,
+          "missed": 22747,
+          "ratio": 0.169727,
+          "total": 27397
+        },
+        "line": {
+          "covered": 827,
+          "missed": 4863,
+          "ratio": 0.145343,
+          "total": 5690
+        },
+        "method": {
+          "covered": 147,
+          "missed": 1052,
+          "ratio": 0.122602,
+          "total": 1199
+        }
+      }
+    },
+    "library_preparation_preflight": {
+      "status": "degraded",
+      "action": "no_action",
+      "issue_number": 2048,
+      "issue_label": "fails-native-image-run",
+      "library": "org.jboss:jandex:2.4.5.Final",
+      "summary": "Library preparation preflight did not produce usable advisory setup.",
+      "deterministic_setup": [],
+      "agent_guidance": "",
+      "risks": [],
+      "applied_setup": [],
+      "input_evidence": [
+        {
+          "name": "issue",
+          "summary": "#2048 [Automation] native-image run fails for org.jboss:jandex:2.4.5.Final; label=fails-native-image-run; body_chars=8015"
+        },
+        {
+          "name": "existing_tests",
+          "summary": "7 existing test file path(s) included"
+        }
+      ],
+      "current_library": "org.jboss:jandex:2.4.2.Final",
+      "new_version": "2.4.5.Final",
+      "failure_reason": "FileNotFoundError: [Errno 2] No such file or directory",
+      "model": "oca/gpt-5.5",
+      "prompt_path": "library-preflight-prompt.txt",
+      "session_log_path": "/home/mihailo/git/graalvm-reachability-metadata/forge/logs/org.jboss:jandex:2.4.5.Final/library-preparation-preflight/pi-generation-2026-06-10T02-00-23-386888Z.log"
+    },
+    "metrics": {
+      "input_tokens_used": 0,
+      "cached_input_tokens_used": 0,
+      "output_tokens_used": 0,
+      "iterations": 0,
+      "cost_usd": 0.0,
+      "tested_library_loc": 836,
+      "metadata_entries": 7,
+      "previous_library_metadata_entries": 3,
+      "code_coverage_percent": 14.45,
+      "previous_library_coverage_percent": 14.53
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.jboss/jandex/2.4.2.Final/src/test/java/org_jboss/jandex/IndexerTest.java",
+      "metadata_file": "metadata/org.jboss/jandex/2.4.5.Final/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/org.jboss/jandex/2.4.5.Final/stats.json
+++ b/stats/org.jboss/jandex/2.4.5.Final/stats.json
@@ -1,0 +1,42 @@
+{
+  "versions" : [ {
+    "version" : "2.4.5.Final",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 3,
+          "totalCalls" : 3
+        },
+        "resources" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 1,
+          "totalCalls" : 1
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 4,
+      "totalCalls" : 4
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 4696,
+        "missed" : 23324,
+        "ratio" : 0.167595,
+        "total" : 28020
+      },
+      "line" : {
+        "covered" : 836,
+        "missed" : 4950,
+        "ratio" : 0.144487,
+        "total" : 5786
+      },
+      "method" : {
+        "covered" : 146,
+        "missed" : 1059,
+        "ratio" : 0.121162,
+        "total" : 1205
+      }
+    }
+  } ]
+}

